modules/json/src/smithy4s/http/json/SchemaVisitorJCodec.scala

```diff
@@ -1157,7 +1157,10 @@ private[smithy4s] class SchemaVisitorJCodec(maxArity: Int)
             stage2 += {
               val value = buffer.get(f.label)
               if (f.isRequired) {
-                if (value == null) cursor.requiredFieldError(f.label, f.label)
+                if (value == null) {
+                  val label = jsonLabel(f)
```

Comment:
Can we avoid recalculation of JSON label in runtime?

Reply:
I looked at it, and it felt super convoluted. I figured that if we end up in this branch, it's happing one and we throw an exception from here, so at this point I think optimizations are out the window

is my assumption correct?

Reply:
You just need to store the following bits of information in an array containing tuples, which you'd use in place of the `fields.foreach` call (line 1156) 

* jsonLabel 
* fieldLabel 
* required

Reply:
made `fields: Fields[Z]` into a fields: LabelledFields[Z]` which is an alias for Vector[(Field[...], String)]` (the string being the json label)

we have label, and required within the field